### ISSUE TITLE
Reflection Issue on Console Commands

### ISSUE THREAD

**Issue by thyseus** [yiisoft/yii2#14129]
### What steps will reproduce the problem?

I have the following console Command:

```php
use yii\console\Controller;
class EPagesController extends Controller
{
    [...]
    public function actionListShops()
    {
        [...]
    }
}
```

Note the second character of the Controller name: uppercase P. I think this is correct "Camelcase".

### What is the expected result?
```

$ ./yii
- epages
    epages/fetch-orders-from-all-shops          Fetches all orders from all shops that uses the EPages REST API.
    epages/fetch-orders-from-shop               Fetches order from specific shop.
    epages/list-shops                           List all Shops that have the Epages REST-Api configured
```

### What do you get instead?

```
$./yii
- epages
- fixture 
```
- every action in epages is /missing/. 

Renaming the controller to EpagesController instead EPagesController fixes the issue. I think the problem is here: https://github.com/yiisoft/yii2/blob/master/framework/console/Controller.php#L566
The Reflection does not handle one-char subparts of an camelcase string properly.

There is more, tough:

Linux:
```
./yii epages/list-shops
Unknown command: epages/list-shops

Did you mean "epages/list-shops"?
[error]           Array content is
{
    "command": "epages/list-shops"
}
```
he does not find the command at all, regardless of lower or uppercase P !

Same on Windows 10 (cygwin bash) works properly but only with lowercase P:
```
./yii epages/list-shops
- works !
```

- Update: It works when i also rename the file to lowercase p. Win10 seems to have case-insensitive files thats why it works under win10... :) - but camelcase issue is still valid in my opinion.

Any Ideas ? Did i spot a tiny bug or am i understanding CamelCase wrong? Thanks a lot!
### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | dev-master
| PHP version      | 7
| Operating system | observed on Win10 and Linux

**Comment by rossoneri** [comment]
I think the best solution is to name your class following the convention！This should not be considered as a bug.

**Comment by vinpel** [comment]
i ran into a similar issue (name a classe camelCase) i see it in the comman line (but no detail), it's not a bug, but a warning message ligne "class need to be respect CamelCase to be a command" could help to find the origine of the problem.

**Comment by SamMousa** [comment]
I think the issue is valid (@rossoneri).
CamelCase merely specify that each word starts with a capital letter.
If those words only have one character than there can be multiple capital letters.

It has to do with the call to `Inflector::camel2id`:
````
 /**
     * Converts a CamelCase name into an ID in lowercase.
     * Words in the ID may be concatenated using the specified character (defaults to '-').
     * For example, 'PostTag' will be converted to 'post-tag'.
     * @param string $name the string to be converted
     * @param string $separator the character used to concatenate the words in the ID
     * @param bool|string $strict whether to insert a separator between two consecutive uppercase chars, defaults to false
     * @return string the resulting ID
     */
    public static function camel2id($name, $separator = '-', $strict = false)
    {
        $regex = $strict ? '/[A-Z]/' : '/(?<![A-Z])[A-Z]/';
        if ($separator === '_') {
            return trim(strtolower(preg_replace($regex, '_\0', $name)), '_');
        } else {
            return trim(strtolower(str_replace('_', $separator, preg_replace($regex, $separator . '\0', $name))), $separator);
        }
    }
````

Combined with line 631 in `Module.php`:
````
$className = str_replace(' ', '', ucwords(str_replace('-', ' ', $className))) . 'Controller';
````

Due to not using strict inflection the inflection is not properly reversed.
I think simply using strict in the case of creating help would work fine.

@thyseus Could you confirm that actually executing a command works even with your controller (just ignore the help).. ?

Basically run `yiic e-pages/list-shops`
